### PR TITLE
Report migration metrics

### PR DESCRIPTION
**Pull request by SupunS**
Closes https://github.com/onflow/cadence/issues/3243

Metrics can be enabled with `--report-metrics` flag.

Currently reports:
- Total values/objects in storage
- Total erroneous values
- Total values/objects related to each contract
- Total erroneous values related to each contract

An "erroneous" value is any value (composite, dictionary, array, capability, etc.) that is associated with a type that is not yet migrated.

### CODE REVIEW

**Review by turbolent — COMMENTED** [review]
Nice work, good idea with the approach 👍

File: cmd/util/ledger/migrations/cadence_values_migration_test.go [hunk 1]
```diff
@@ -224,12 +224,12 @@ func checkMigratedState(
 			{
 				storageKey:    interpreter.StorageKey{Key: "storage", Address: address},
 				storageMapKey: interpreter.StringStorageMapKey("flowTokenVault"),
-				value:         `11240984669916758018`,
+				value:         `8791026472627208194`,
```
[the remark this answers is not in the document]
Comment (SupunS) [reply cmd/util/ledger/migrations/cadence_values_migration_test.go:227]: I updated the emulator state, so the old `Test` contract deployed for tests is not compatible with Cadence 1.0.

[the remark this answers is not in the document]
Comment (turbolent) [reply cmd/util/ledger/migrations/cadence_values_migration_test.go:227]: Ah, I see, missed that 👍 Did you manually modify the state file or re-generate it. If it is the latter, could you please also update the files in https://github.com/onflow/flow-go/tree/supun/migration-stats/cmd/util/ledger/migrations/test-data/cadence_values_migration to produce that state?

**Review by turbolent — APPROVED** [review]
👌

File: cmd/util/ledger/migrations/migration_metrics_collector.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+func (c *MigrationMetricsCollector) RecordValueForContract(location common.Location) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	c.ValuesPerContract[location]++
+}
+
+func (c *MigrationMetricsCollector) metrics() Metrics {
+	errorsPerContract := make(map[string]int, len(c.ErrorsPerContract))
```
[the remark this answers is not in the document]
Comment (SupunS) [reply cmd/util/ledger/migrations/migration_metrics_collector.go:351]: Added https://github.com/onflow/flow-go/pull/5988/commits/3619d9abffe684d3cdcdb1fab391e4d847fd90fb

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**